website/addons/onedrive/views.py

```diff
@@ -0,0 +1,191 @@
+"""Views for the node settings page."""
```

Comment:
The code here in general looks very solid. 

We did just release some new addon code using generic view method factories: https://github.com/CenterForOpenScience/osf.io/blob/develop/website/addons/base/generic_views.py (with an example usage: https://github.com/CenterForOpenScience/osf.io/blob/develop/website/addons/dropbox/views.py)

Certainly not a blocker for this PR, but you might consider using the generic views whenever possible. One advantage to this is you get lots of free tests ([see dropbox](https://github.com/CenterForOpenScience/osf.io/blob/develop/website/addons/dropbox/tests/test_views.py)), and you help us improve uniformity across the addon code. 

But again, since you've already taken the time to write these views and corresponding tests, this is optional.